worker/peergrouper/worker_test.go

```diff
@@ -441,7 +441,7 @@ func (s *workerSuite) TestControllersArePublishedOverHub(c *gc.C) {
 
 	expected := apiserver.Details{
 		Servers: map[string]apiserver.APIServer{
-			"10": apiserver.APIServer{ID: "10", Addresses: []string{"0.1.2.10:5678"}},
+			"10": apiserver.APIServer{ID: "10", Addresses: []string{"0.1.2.10:5678"}, InternalAddress: "0.1.2.10:5678"},
 			"11": apiserver.APIServer{ID: "11", Addresses: []string{"0.1.2.11:5678"}},
```

Comment:
Only the first one is a voting member of the replicaset, and - at the moment - the peergrouper only includes those servers that are currently voting members in Mongo. It's not ideal, bit it's KISS. It would probably be better to have the raftclusterer watch the HA configuration directly, and side step peergrouper altogether.